### PR TITLE
Add a document describing AMD NPU workings

### PR DESCRIPTION
**Pull request by sonals**

### CODE REVIEW

File: src/driver/doc/amdnpu.rst [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+The microcontroller and amdxdna driver use a dedicated user channel per
+workload context. The user channel is primarily used for submitting work to
+the NPU. As mentioned before, a user channel requests are serviced by an
+instance of ERT.
+
+PCIe EP
+-------
+
+NPU is visible to the x86 as a PCIe device with 3 BARS and an MSI-X interrupt
```
Comment (maxzhen) [inline src/driver/doc/amdnpu.rst:79]: an MSI-X interrupt -> some MSI-X interrupts? We have one interrupt per context + one for mert.

File: src/driver/doc/amdnpu.rst [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+The open-source IREE compiler supports graph compilation of ML models for AMD
+NPU and uses Peano underneath. It is available at:
+https://github.com/nod-ai/iree-amd-aie
+
+Usermode Driver (UMD)
+---------------------
+
+The open-source XRT runtime stack interfaces with amdxdna kernel driver. XRT
+can be found at:
+https://github.com/Xilinx/XRT
```
Comment (maxzhen) [inline src/driver/doc/amdnpu.rst:220]: Do you want to mention this repo? People asked about it before...

File: src/driver/doc/amdnpu.rst [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+Application Binaries
+====================
+
+A NPU application workload is comprised of two separate binaries which are
+generated by the NPU compiler.
+
+1. AMD XDNA Array overlay, which is used to configure a NPU spatial partition.
+   The overlay contains instructions for setting up the stream switch
+   configuration and ELF for the compute tiles. The overlay is loaded by the
+   microcontroller on the spatial partition when the workload context becomes
```
Comment (pranjalv-xilinx) [inline src/driver/doc/amdnpu.rst]: will be good to mention that overlay is also loaded by ERT (running in protected mode) on microcontroller

File: src/driver/doc/amdnpu.rst (hunk 2 above)
Comment (sonals) [reply src/driver/doc/amdnpu.rst:220]: Yes, will update

File: src/driver/doc/amdnpu.rst (hunk 3 above)
Comment (sonals) [reply src/driver/doc/amdnpu.rst]: Yes, will update

File: src/driver/doc/amdnpu.rst (hunk 1 above)
Comment (sonals) [reply src/driver/doc/amdnpu.rst:79]: Updated

**Comment by sonals** [comment]
ok to test

File: src/driver/doc/amdnpu.rst (hunk 2 above)
Comment (sonals) [reply src/driver/doc/amdnpu.rst:220]: Done

**Comment by maxzhen** [comment]
> ok to test

The CI can't be triggered here. Since this is a doc only update, we can ignore CI.

File: src/driver/doc/amdnpu.rst [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+called ERT to service each workload context. ERT is also used to execute user
+provided ``ctrlcode`` associated with the workload context.
+
+NPU Firmware uses a single isolated privileged context called MERT to service
+management commands from the amdxdna driver.
+
+Mailboxes
+.........
+
+The microcontroller and amdxdna driver use a privileged channel for management
```
Comment (keryell) [inline src/driver/doc/amdnpu.rst:66]:
```suggestion
The microcontroller and ``amdxdna`` driver use a privileged channel for management
```
everywhere

**Review by keryell — APPROVED** [review]
Great documentation !!!

File: src/driver/doc/amdnpu.rst (hunk 4 above)
Comment (sonals) [reply src/driver/doc/amdnpu.rst:66]: Should we use bold or monospace?

**Pull request merged**